source/LoadPanel.cpp

```diff
@@ -456,7 +456,7 @@ void LoadPanel::SnapshotCallback(const string &name)
 
 	// If a file with this name already exists, make sure the player
 	// actually wants to overwrite it.
-	string to = from.substr(0, from.size() - 4) + extension;
+	string to = Files::Saves() + selectedPilot + extension;
```

Comment:
Your proposed change is exactly equivalent to the current value based on my testing -- can you provide a step-by-step reproduction in which they will not be the same?

Reply:
1. Open the load panel
2. Click the "Add snapshot" button
3. Type something (or leave the date there)
4. Click the "OK" button; then do not load the game
5. Click the "Add snapshot" button
6. Type something (or leave the date there)
7. Click the "OK" button

Reply:
That is the exact reproduction process I was performing. There is no difference between the original value and your proposed value.

I'm curious -- when you hover the snapshot created in step 4, what is the created timestamp? Is it the same as the source pilot? or different?

For me:
![image](https://user-images.githubusercontent.com/20871346/148686170-3eb11dd8-661b-464b-aaa3-af9eb6e192b3.png)
![image](https://user-images.githubusercontent.com/20871346/148686188-adabf3ba-c94e-4f0f-a34d-c27f1d50262e.png)
![image](https://user-images.githubusercontent.com/20871346/148686208-bfc5c4a9-7d01-458f-8e6c-7b7b197f8627.png)

Reply:
> I'm curious -- when you hover the snapshot created in step 4, what is the created timestamp? Is it the same as the source pilot? or different?

It is later than original one.

Reply:
I think I would prefer to fix that issue -- `Files::Copy` not preserving the source files' timestamps -- than address only the names.

That has the side effect of fixing any other future uses of `Files::Copy` in the program to have the same behavior across platforms, and also results in a consistent ordering of the player's pilot file and snapshots across platforms.